onnxruntime/core/session/inference_session.h

```diff
@@ -383,6 +383,8 @@ class InferenceSession {
 
   // The file path of where the model was loaded. e.g. /tmp/test_squeezenet/model.onnx
   std::basic_string<ORTCHAR_T> model_location_;
+  
+  SessionOptions session_options_;
 
```

Comment:
Why does this need to be moved from private to protected? I think all the members here can be private (was fine in the build I tested but the settings may be different). If a derived class needs access add a getter.